sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java

```diff
@@ -185,32 +211,330 @@ HttpPipeline getHttpPipeline() {
+                .doOnSuccess(response ->
+                    logger.verbose("Listed next role definitions page - Page {}", continuationToken))
+                .doOnError(error ->
+                    logger.warning("Failed to list next role definitions page - Page {}", continuationToken, error))
+                .onErrorMap(KeyVaultAdministrationUtils::mapThrowableToKeyVaultAdministrationException)
                 .map(KeyVaultAccessControlAsyncClient::transformRoleDefinitionsPagedResponse);
         } catch (RuntimeException e) {
             return monoError(logger, e);
         }
     }
 
+    /**
+     * Creates or updates a {@link KeyVaultRoleDefinition} with a randomly generated {@link String name}.
+     *
+     * @param roleScope The {@link KeyVaultRoleScope role scope} of the {@link KeyVaultRoleDefinition}. Managed HSM
+     * only supports '/'.
+     *
+     * @return A {@link Mono} containing the created {@link KeyVaultRoleDefinition}.
+     *
+     * @throws KeyVaultAdministrationException If the given {@code roleScope} is invalid.
+     * @throws NullPointerException If the {@link KeyVaultRoleScope role scope} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<KeyVaultRoleDefinition> setRoleDefinition(KeyVaultRoleScope roleScope) {
+        return setRoleDefinition(roleScope, UUID.randomUUID().toString());
+    }
+
+    /**
+     * Creates or updates a {@link KeyVaultRoleDefinition}.
+     *
+     * @param roleScope The {@link KeyVaultRoleScope role scope} of the {@link KeyVaultRoleDefinition}. Managed HSM only
+     * supports '/'.
+     * @param roleDefinitionName The name of the {@link KeyVaultRoleDefinition}. It can be any valid UUID.
+     *
+     * @return A {@link Mono} containing the created {@link KeyVaultRoleDefinition}.
+     *
+     * @throws KeyVaultAdministrationException If the given {@code roleScope} is invalid.
+     * @throws NullPointerException If the {@link KeyVaultRoleScope role scope} or {@link String roleDefinitionName}
+     * are {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<KeyVaultRoleDefinition> setRoleDefinition(KeyVaultRoleScope roleScope, String roleDefinitionName) {
+        return setRoleDefinitionWithResponse(new SetKeyVaultRoleDefinitionOptions(roleScope, roleDefinitionName))
+            .flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Creates or updates a {@link KeyVaultRoleDefinition}.
+     *
+     * @param options Object representing the configurable options to create or update a
+     * {@link KeyVaultRoleDefinition role definition}.
+     *
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains the
+     * created or updated {@link KeyVaultRoleDefinition}.
+     *
+     * @throws KeyVaultAdministrationException If any parameter in {@code options} is invalid.
```

Comment:
Shouldn't that class throw in that case? Besides, we should be careful about too much client-side validation. Validating non-null for required path parameters is fine, but that's about it beyond type verification/safety. Guidelines around parameters are available in our public guidelines site.

Reply:
I see just below the validations you're doing. Those are probably fine, but I leave this here such as a general note if nothing else.